### PR TITLE
Public Test Net Updates

### PR DESCRIPTION
**Pull request by ejfitzgerald**
- Quick fix for HEAD block size issue
- Fix HTTP empty buffer crash
- Restrict miner
- Minor logging updates
- Environment variable configuration support
- P2P manifest fix

### CODE REVIEW

File: apps/constellation/main.cpp [hunk 1]
```diff
@@ -385,6 +495,9 @@ ProverPtr GenerateP2PKey()
 {
   static constexpr char const *KEY_FILENAME = "p2p.key";
```
Comment (wkaluza) [inline apps/constellation/main.cpp:496]: Maybe combine KEY_FILENAME and key_path into a single std::string variable? Given that we still need to access key_path.c_str() later, we might as well.

Relatedly, is key_path actually a path? Or should it be called a file name?

File: apps/constellation/main.cpp [hunk 2]
```diff
@@ -183,6 +228,33 @@ struct CommandLineArguments
     // parse the args
     p.Parse(argc, argv);
 
+    // if the user has specified any environment variables these always take precedence
```
Comment (pbukva) [inline apps/constellation/main.cpp:231]: Should this be other way around? As per well esstablished approach, command-line arguments should take precedence over Env. variables (or config files in that matter).

File: libs/ledger/src/chain/main_chain.cpp [hunk 3]
```diff
@@ -489,11 +489,14 @@ void MainChain::RecoverFromFile(Mode mode)
   if (Mode::CREATE_PERSISTENT_DB == mode)
   {
     block_store_->New("chain.db", "chain.index.db");
+    head_store_.open("chain.head.db",
```
Comment (wkaluza) [inline libs/ledger/src/chain/main_chain.cpp:492]: Does this file not potentially leak the head_store_?

**Review by pbukva — APPROVED** [review]
LGTM

File: apps/constellation/main.cpp (hunk 1 above)
Comment (ejfitzgerald) [reply apps/constellation/main.cpp:496]: It is a path, it just can't be a folder path.

Comment (ejfitzgerald) [reply apps/constellation/main.cpp:496]: I tend to agree about the structure I will raise an issue to cover this and the @pbukva comments

File: apps/constellation/main.cpp (hunk 2 above)
Comment (ejfitzgerald) [reply apps/constellation/main.cpp:231]: Raise #915 to track this

**Pull request merged**